modules/visualisation/src/backend/api.ts

```diff
@@ -0,0 +1,78 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import ConfusionMatrix from 'ml-confusion-matrix'
+import { Matrix } from 'ml-matrix'
+import nanoid from 'nanoid'
+
+import { getTrainTestDatas } from '../tools/data_loader'
+import {
+  computeConfusionMatrix,
+  computeEmbeddingSimilarity,
+  computeIntentSimilarity,
+  computeOutliers,
+  computeScatterEmbeddings
+  // computeTsneScatterEmbeddings
+} from '../tools/visualisation'
+
+import { VisuState } from './typings'
+
+export default async (bp: typeof sdk, state: VisuState) => {
+  const longJobsPool = {}
+  const router = bp.http.createRouterForBot('new_qna')
+  const glob_res = []
+
+  router.get('/confusionMatrix', async (req, res) => {
+    const botId = req.params.botId
+    const newAxiosConfig = await bp.http.getAxiosConfigForBot(botId, { localUrl: true })
+    state[botId].predictor.axiosConfig = newAxiosConfig
+    state[botId].axiosConfig = newAxiosConfig
+    const jobId = nanoid()
+    res.send(jobId)
+    longJobsPool[jobId] = { status: 'computing', data: undefined, error: undefined, cm: true }
+    try {
+      longJobsPool[jobId].data = await computeConfusionMatrix(state[botId], glob_res)
+      longJobsPool[jobId].status = 'done'
+      console.log('\n\n Done CM \n\n')
```

Comment:
Can you use an alias instead? Also, you don't have to declare it to undefined, simply leaving them out would do the trick.
 ```
const job = longJobsPool[jobId]
job.status  = 'done'```js